transaction/src/validation/transaction_validator.rs

```diff
@@ -888,4 +888,30 @@ mod tests {
             ))
         );
     }
+
+    #[test]
+    fn test_clone_proof() {
+        let transaction = create_transaction_advanced(
+            Epoch::of(0),
+            Epoch::of(40),
+            123,
+            vec![55],
+            66,
+            ManifestBuilder::new()
+                .take_from_worktop(XRD, dec!(100), "bucket")
+                .create_proof_from_bucket_of_amount("bucket", dec!(5), "proof")
+                .then(|builder| {
+                    let lookup = builder.name_lookup();
+                    let proof_id = lookup.proof("proof");
+                    builder
```

Comment:
Given this test name, I'd expect this to use the clone proof command, rather than advanced instructions.

Perhaps it would be a more explicit to create a test file for the manifest builder itself, and explicitly add tests for the "add_instruction_advanced" command, which checks it is collision resistant against existing proofs/buckets named "proof" and "bucket"?